tests/tests.rs

```diff
@@ -1458,7 +1458,7 @@ fn test_max_results() {
     );
 
     // Limited to one result. We could find either C.Foo2 or c.foo
-    let output = te.assert_success_and_get_output(".", &["--max-results=1", "c.foo"]);
```

Comment:
I would like to keep this test written with `--max-results=1`. If we want to add a test for the new option (which definitely doesn't hurt), I suggest we move these few lines into a closure or new function and call it twice, with `--max-results=1` and `-1`.

Reply:
I have done so with a closure.